### ISSUE TITLE
Maybe no need to duplicate the entire Compose file?

### ISSUE THREAD

**Issue by kriansa** [marcopaganini/gnome-cedilla-fix#6]
Hi @marcopaganini ,

I've spent some time on this issue since it started to annoy me this week. I ended up with a smaller solution that you might be interested in testing: https://garajau.com.br/2021/02/enabling-cedilla-acute-c-on-gnome

Basically, you don't need to copy the entire Compose file, just the overrides. Moreover,  while the sole `.XCompose` file will work fine for non-GTK applications, when on GTK apps, the overrides won't pick up, unless I explicitly tell GNOME to use `ibus` as the input method.

Let me know if that works for you! :+1:

**Comment by marcopaganini** [comment]
Hello @kriansa

This is correct, and in fact, I used a configuration very similar to yours the beginning. The point is that, for me, I'd prefer to save all my current customizations (hence the trick of copying the original Compose file and changing things in there). I understand that it may not be what most people may want. I'll maybe add the option to save only the overrides.

**Comment by marcopaganini** [comment]
[Hit Send too soon]

Regarding ibus, for me it works with or without ibus. What does your `GTK_IM_MODULE` environment variable contains?

**Comment by kriansa** [comment]
That's the point, I'm not setting `GTK_IM_MODULE` at all - I'd rather not touch `/etc/environment` to keep it clean and under my dotfiles control.

In the past I just had the .XCompose overrides and it worked just fine. However I use arch and very recently I noticed it stopped working - can't tell which package "broke" it, but now I'm forced to use `ibus`. But since I've been forced into it, I'd rather simply not set `GTK_IM_MODULE` globally, that's why I spent some time after a solution that wouldn't require modifying system files.

**Comment by marcopaganini** [comment]
Yeah, I agree. If you notice, we don't touch `/etc/environment` anymore. It's really not needed. In the past we needed the "cedilla" module, but at some point that stopped working. Now it's basically xim or ibus and both seem to work for me at least. One weird exception is Telegram (Qt), which "loses" accents after a few minutes of use. Switching windows fixes the problem. I'm investigating that but it's a difficult to find the cause for that.
